src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt

```diff
@@ -98,7 +98,7 @@ class Cargo(
     fun reformatFile(owner: Disposable, file: VirtualFile, listener: ProcessListener? = null): ProcessOutput {
         val cmd = CargoCommandLine(
             "fmt", file.parent.pathAsPath,
-            listOf("--all", "--", "--write-mode=overwrite", "--skip-children", file.path)
+            listOf("--all", "--", "--skip-children", file.path)
```

Comment:
We can't just drop this parameter because `write-mode` flag is dropped only in nightly builds
Let's check that current rustfmt don't support `write-mode` and drop this flag only in this case
You can do it in a similar way as we do for `all-targets` flag of `cargo check` command (see [here](https://github.com/intellij-rust/intellij-rust/blob/master/src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt#L112-L113))
Also, looks like that it makes sense to use `--emit=files` flag to commit behaviour of `rustfmt`

Reply:
updated. 
`rustfmt` is used instead of `cargo fmt` because https://github.com/rust-lang-nursery/rustfmt/pull/2768 , and I don't know how to get the project root as workingDirectory easily. 
support of `write-mode` and `skip-children` is checkd